clang-tools-extra/clang-tidy/daedalean/TypeConversionsCheck.cpp

```diff
@@ -207,74 +207,74 @@ void TypeConversionsCheck::handleImplicitCast(clang::ASTContext *context,
                                               QualType sourceType,
                                               QualType destType,
                                               SourceLocation location) {
-  sourceType = sourceType.getCanonicalType();
-  destType = destType.getCanonicalType();
-  if (destType == sourceType) {
+  const QualType canonicalSourceType = sourceType.getCanonicalType();
```

Comment:
Should we handle static cast in the same manner?

Reply:
No idea, I could not make test of small size even for this

Reply:
Ok, let's not do it for now and we can handle it later if needed.